### PR TITLE
Henryh/dino v2 eval

### PR DESCRIPTION
**Pull request by Hgherzog**
Running Evals with Panopticon and DinoV2

- Added EvalWrapper class to have a common interface for evals depending on the model being used

Also found some bugs along the way
- Fix bugs for dataset based normalization for pastis
- Fix dataset normalization bug in so2sat with 10k multiply

### CODE REVIEW

File: scripts/2025_06_26_dataset_percentage_experiments/eval.py
Comment (Hgherzog) [inline scripts/2025_06_26_dataset_percentage_experiments/eval.py:1]: I will revert the changes in this folder it was for debugging

File: helios/evals/dinov2/dinov2.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
+"""DINOv2 model wrapper."""
```
Comment (gabrieltseng) [inline helios/evals/dinov2/dinov2.py]: can dinov2 and panopticon be moved into `helios/evals/models`?

This will prevent `helios/evals` from getting too cluttered, especially as we add more models.

File: helios/evals/dinov2/dinov2.py [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+
+def make_normalize_transform(
+    mean = IMAGENET_DEFAULT_MEAN,
+    std = IMAGENET_DEFAULT_STD,
+) -> transforms.Normalize:
+    return transforms.Normalize(mean=mean, std=std)
+
+HELIOS_SENTINEL2_RGB_BANDS = [3, 2, 1]
+HELIOS_LANDSAT_RGB_BANDS = [4, 3, 2]
```
Comment (gabrieltseng) [inline helios/evals/dinov2/dinov2.py]: Could this be linked to `Modality` instead, so that it doesn't break if we change things?

I think it would look like

```python
HELIOS_SENTINEL2_RGB_BANDS = [Modality.SENTINEL2_L2A.band_order.index(b) for b in ["B04", "B03", "B02"]]
```
I also think `HELIOS_SENTINEL2_RGB_BANDS` should be `[2, 1, 0]`, not `[3, 2, 1]` since B1 is later in the band order:

```python
>>> from helios.data.constants import Modality
>>> Modality.SENTINEL2_L2A.band_order
['B02', 'B03', 'B04', 'B08', 'B05', 'B06', 'B07', 'B8A', 'B11', 'B12', 'B01', 'B09']
>>> Modality.SENTINEL2.band_order
['B02', 'B03', 'B04', 'B08', 'B05', 'B06', 'B07', 'B8A', 'B11', 'B12', 'B01', 'B09', 'B10']
```

File: helios/evals/dinov2/dinov2.py [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+        per_timestep_inputs = []
+        for i, input_data_i in input_data_timesteps.items():
+            # Concatenate all modality data along channel dimension
+            concatenated_imgs = torch.cat(input_data_i, dim=1)
+
+            per_timestep_inputs.append(concatenated_imgs)
+        return per_timestep_inputs
+
+    # pooling type is on the timesteps only right now
+    def forward(self, masked_helios_sample: MaskedHeliosSample, pooling: PoolingType = PoolingType.MEAN, apply_imagenet_normalization: bool = False) -> torch.Tensor:
```
Comment (gabrieltseng) [inline helios/evals/dinov2/dinov2.py]: nit - `apply_imagenet_normalization` may be better as an argument to `__init__` so that we can control it from the config.

File: helios/evals/embeddings.py [hunk 4]
```diff
@@ -1,31 +1,29 @@
 """Embeddings from models."""
 
 import logging
-
+import math
```
Comment (gabrieltseng) [inline helios/evals/embeddings.py]: This import isn't used. I feel like `ruff` should have caught this?

Comment (gabrieltseng) [reply helios/evals/embeddings.py]: Same with `import torch.nn.functional as F` below.

Comment (gabrieltseng) [reply helios/evals/embeddings.py]: And `einops`, and `Modality`.

File: helios/evals/eval_wrapper.py [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+
+class EvalWrapper:
+    def __init__(self, model: nn.Module, task_type: TaskType, patch_size: int, pooling_type: PoolingType, concat_features: bool = False, apply_imagenet_normalization: bool = False):
+        super().__init__()
+        self.model = model
+        self.task_type = task_type
+        self.patch_size = patch_size
+        self.pooling_type = pooling_type
+        self.concat_features = concat_features
+        self.apply_imagenet_normalization = apply_imagenet_normalization
```
Comment (gabrieltseng) [inline helios/evals/eval_wrapper.py]: this seems like a very specific kwarg to have in this general wrapper. I think only Dino and Panopticon will make use of them ; maybe it makes sense as a model specific kwarg (as suggested [here](https://github.com/allenai/helios/pull/299/files#r2242354491)).

File: helios/evals/panopticon/panopticon.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+        patch_size: int = 14,
+    ):
+        """Initialize the Panopticon wrapper.
+
+        Args:
+            torchhub_id: The torch hub model ID for panopticon
+            patch_size: Patch size for the vision transformer (default 14)
+        """
+        super().__init__()
+        self.patch_size = patch_size
```
Comment (gabrieltseng) [inline helios/evals/panopticon/panopticon.py]: I don't think Panopticon supports variable patch sizes; should we throw a `ValueError` if patch size is not 14?

File: helios/evals/dinov2/dinov2.py [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+                    align_corners=False
+                )
+            else:
+                # Resize the image to the dinov2 pre-training input size
+                image_size = 224
+                data_i = F.interpolate(
+                    data_i,
+                    size=(image_size, image_size),
+                    mode="bilinear",
+                    align_corners=False
```
Comment (gabrieltseng) [inline helios/evals/dinov2/dinov2.py]: nit - here and in `panopticon.py`, this code block could be simplified:

```python
new_height = self.patch_size if original_height == 1 else 224

data_i = F.interpolate(
        data_i,
        size=(new_height, new_height),
        mode="bilinear",
        align_corners=False
    )
```

File: helios/evals/embeddings.py (hunk 4 above)
Comment (Hgherzog) [reply helios/evals/embeddings.py]: I need to run pre-commit will take care of all of this.

File: helios/evals/dinov2/dinov2.py (hunk 2 above)
Comment (Hgherzog) [reply helios/evals/dinov2/dinov2.py]: Good catch.

File: helios/evals/eval_wrapper.py (hunk 5 above)
Comment (Hgherzog) [reply helios/evals/eval_wrapper.py]: it only even applies to dinov2

File: helios/evals/panopticon/panopticon.py (hunk 6 above)
Comment (Hgherzog) [reply helios/evals/panopticon/panopticon.py]: I just made patch size a class variable because it is a constant for these models
